### DISCUSSION TITLE
"OSI model" for SMR (from team week Q3'22 Istanbul)

### DISCUSSION THREAD

**Discussion by matejpavlovic** [consensus-shipyard/consensuslab#173] [General]
Create a set of abstractions generally describing the components of an SMR system.
See [presentation](https://docs.google.com/presentation/d/1B2YgFcvRJeBckxRcvFXqQT_E4YnC0FWAZTUDlKzEIPg/edit?usp=sharing) for more details.

**Comment by LefKok** [comment]
We can extend the work we started here on modularising blockchains with @guy-goren https://eprint.iacr.org/2022/455
